pkg/rancher-desktop/pages/volumes/files/_name.vue

```diff
@@ -0,0 +1,536 @@
+    };
+  },
+  computed: {
+    ...mapGetters('k8sManager', {isK8sReady: 'isReady'}),
+    volumeName() {
+      const name = this.$route.params.name;
+      if (!name || !/^[a-zA-Z0-9._-]+$/.test(name)) {
+        throw new Error('Invalid volume name format');
+      }
+      return name;
+    },
+    hasNamespaceSelected() {
+      return this.settings?.containerEngine?.name === ContainerEngine.CONTAINERD && this.settings?.containers?.namespace;
+    },
+    pathSegments() {
+      return this.currentPath
+        .split('/')
+        .filter(segment => segment !== '');
+    },
+  },
+  watch: {
+    '$route.query.path': {
+      handler(newPath) {
+        const path = newPath || '/';
+        if (this.currentPath !== path) {
+          this.currentPath = path;
+          if (this.ddClient && this.volumeExists) {
+            this.isLoading = true;
+            this.listFiles();
+          }
+        }
+      },
+      immediate: false
+    }
+  },
+  mounted() {
+    this.$store.dispatch('page/setHeader', {
+      title: this.t('volumes.files.title'),
+      description: this.volumeName,
+    });
+
+    ipcRenderer.on('settings-read', this.onSettingsRead);
+    ipcRenderer.send('settings-read');
+
+    this.refreshInterval = setInterval(() => {
+      if (!this.isLoading && this.volumeExists) {
+        this.listFiles();
+      }
+    }, 30000);
+  },
+  beforeDestroy() {
+    ipcRenderer.off('settings-read', this.onSettingsRead);
+    if (this.refreshInterval) {
+      clearInterval(this.refreshInterval);
+    }
+  },
+  methods: {
+    async onSettingsRead(event, settings) {
+      this.settings = settings;
+      await this.initializeFileBrowser();
+    },
+    async initializeFileBrowser() {
+      if (window.ddClient && this.isK8sReady && this.settings) {
+        this.ddClient = window.ddClient;
+        await this.checkVolumeExists();
+        if (this.volumeExists) {
+          await this.listFiles();
+        }
+      }
+    },
+    async checkVolumeExists() {
+      try {
+        const options = {};
+        if (this.hasNamespaceSelected) {
+          options.namespace = this.hasNamespaceSelected;
+        }
+
+        const volumes = await this.ddClient?.docker.listVolumes(options);
+        this.volumeExists = volumes?.some(v => v.Name === this.volumeName) || false;
+
+        if (!this.volumeExists) {
+          this.error = this.t('volumes.files.volumeNotFound', {name: this.volumeName});
+        }
+      } catch (error) {
+        console.error('Error checking volume:', error);
+        this.volumeExists = false;
+        this.error = this.t('volumes.files.checkError');
+      }
+    },
+    async listFiles() {
+      try {
+        this.error = null;
+
+        const containerPath = `/volume${this.currentPath}`;
+        const lsCommand = [
+          'run', '--rm',
+          '-v', `${this.volumeName}:/volume:ro`,
+          'busybox',
+          'sh', '-c',
+          `cd "${containerPath}" && ls -la | tail -n +2`
```

Comment:
Consider not running this through `sh` and make `parseLsOutput` handle the first line instead.